### PR TITLE
Added Storefront API Posts and Post Categories endpoints

### LINKED ISSUE #13047: Add Posts and Post categories to Storefront API
We need API endpoints in the Storefront API to fetch Posts/post categories.

### PR DESCRIPTION
**Pull request by damianlegawiec**
Fixes #13047

## Summary by CodeRabbit

- **New Features**
  - Storefront API endpoints to list/view posts and post categories; show can include associated published posts.
  - Posts support rich fields (excerpt, content, metadata, tags, image URL), category relationship, search, filtering (category, author, IDs, slugs, tags), and multiple sort options (including by published date).
  - Public read access enabled for posts and post categories.
- **Tests**
  - Added comprehensive request specs covering index/show, filters, search, sorting, and edge cases.
- **Chores**
  - Updated local settings schema reference and expanded tool permissions.

### CODE REVIEW

File: core/app/models/spree/ability.rb [hunk 1]
```diff
@@ -101,6 +101,8 @@ def apply_user_permissions(user, _options)
       can :accept, Spree::Invitation, invitee_id: [user.id, nil], invitee_type: user.class.name, status: 'pending'
       can :read, ::Spree::Policy
       can :read, ::Spree::Page
+      can :read, ::Spree::Post
+      can :read, ::Spree::PostCategory
```
[the remark this answers is not in the document]
Comment (damianlegawiec) [reply core/app/models/spree/ability.rb:105]: we don't need that, that is taken care of by ResourceController automatically
